arcgis-ios-sdk-samples/Edit data/Add features with contingent values/AddFeaturesContingentValuesViewController.swift

```diff
@@ -0,0 +1,212 @@
+        switch result {
+        case .success:
+            // Get and load the first feature table in the geodatabase.
+            let featureTable = geodatabase.geodatabaseFeatureTables[0]
+            self.featureTable = featureTable
+            featureTable.load { [weak self] error in
+                guard let self = self else { return }
+                if let error = error {
+                    self.presentAlert(error: error)
+                } else {
+                    // Create and load the feature layer from the feature table.
+                    let featureLayer = AGSFeatureLayer(featureTable: featureTable)
+                    // Add the feature layer to the map.
+                    self.mapView.map?.operationalLayers.add(featureLayer)
+                    // Set the map's viewpoint to the feature layer's full extent.
+                    let extent = featureLayer.fullExtent
+                    self.mapView.setViewpoint(AGSViewpoint(targetExtent: extent!))
+                    // Add buffer graphics for the feature layer.
+                    self.queryFeatures()
+                }
+            }
+        case .failure(let error):
+            presentAlert(error: error)
+        }
+    }
+    
+    /// Create a map with a vector tiled layer basemap.
+    func makeMap() -> AGSMap {
+        // Get the vector tiled layer by name.
+        let fillmoreVectorTiledLayer = AGSArcGISVectorTiledLayer(name: "FillmoreTopographicMap")
+        // Use the vector tiled layer as a basemap.
+        let basemap = AGSBasemap(baseLayer: fillmoreVectorTiledLayer)
+        let map = AGSMap(basemap: basemap)
+        return map
+    }
+    
+    /// Create buffer graphics for the features.
+    func queryFeatures() {
+        guard let featureTable = featureTable else { return }
+        // Create the query parameters.
+        let queryParameters = AGSQueryParameters()
+        // Set the where clause to filter for buffer sizes greater than 0.
+        queryParameters.whereClause = "BufferSize > 0"
+        // Query the features with the query parameters.
+        featureTable.queryFeatures(with: queryParameters) { [weak self] result, error in
+            guard let self = self else { return }
+            if let error = error {
+                // Present an alert if there is an error while querying the features.
+                self.presentAlert(error: error)
+            } else if let result = result {
+                // Creates buffer symbol for each feature according to its buffer size.
+                let newGraphics = result.featureEnumerator().allObjects.map(self.makeGraphic(feature:))
+                // Replace existing graphics with the new graphics.
+                self.graphicsOverlay.graphics.setArray(newGraphics)
+            }
+        }
+    }
+    
+    /// Add a single feature to the map.
+    func addFeature(at mapPoint: AGSPoint) {
+        // Create a symbol to represent a bird's nest.
```

Comment:
Might be worth checking if the mapPoint is within the map package's extent. Currently I can add features outside the basemap visible area.

Reply:
Good idea. How can I check?

Reply:
Sth like below, but you might need to improvise as I've never done this before 😅 
```swift
guard let mapExtent = fillmoreVectorTiledLayer.fullExtent, AGSGeometryEngine.geometry(mapPoint, within: mapExtent) else { return }
```
I'm OK without this change.